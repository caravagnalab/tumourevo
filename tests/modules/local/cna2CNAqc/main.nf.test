nextflow_process {
    name "Test Process CNA_PROCESSING (cna2CNAqc)"
    script "modules/local/cna2CNAqc/main.nf"
    process "CNA_PROCESSING"

    test("Should run without failures") {
        when {
            params {
                // define module specific parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                meta = [
                    id : "MSeq_Set06_Set6_42",
                    dataset : "MSeq",
                    patient : "Set06",
                    tumour_sample: "Set6_42",
                    cancer_type: "PANCANCER" ,
                    cna_caller : "sequenza"
                ]
                cna_segs = file(params.local_testdata_base_path + '/from_input/Set6_42.smoothedSegs.txt', checkIfExists: true)
                cna_extra = file(params.local_testdata_base_path + '/from_input/Set6_42_confints_CP.txt', checkIfExists: true)
                input[0] = tuple(meta, cna_segs, cna_extra)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
