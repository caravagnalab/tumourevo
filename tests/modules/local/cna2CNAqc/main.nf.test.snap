{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "MSeq_Set06_Set6_42",
                            "dataset": "MSeq",
                            "patient": "Set06",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER",
                            "cna_caller": "sequenza"
                        },
                        "MSeq_Set06_Set6_42_cna.rds:md5,3274b43f2743dc170d5957329d64b455"
                    ]
                ],
                "rds": [
                    [
                        {
                            "id": "MSeq_Set06_Set6_42",
                            "dataset": "MSeq",
                            "patient": "Set06",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER",
                            "cna_caller": "sequenza"
                        },
                        "MSeq_Set06_Set6_42_cna.rds:md5,3274b43f2743dc170d5957329d64b455"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-08-29T17:12:09.579553143"
    }
}